modules/handPointing/handPointingThread.cpp

```diff
@@ -0,0 +1,544 @@
+    bool r_shoulder_frame_exists = m_iTc->getTransform(m_r_shoulder_frame_id, m_base_frame_id, m_r_transform_mtrx_shoulder);
+    if (!r_shoulder_frame_exists)
+    {
+        yCWarning(HAND_POINTING_THREAD, "Unable to found m matrix (r_shoulder-base)");
+    }
+
+    // compute the transformation matrix from the left shoulder to the base reference frame
+    bool l_shoulder_frame_exists = m_iTc->getTransform(m_l_shoulder_frame_id, m_base_frame_id, m_l_transform_mtrx_shoulder);
+    if (!l_shoulder_frame_exists)
+    {
+        yCWarning(HAND_POINTING_THREAD, "Unable to found m matrix (l_shoulder-base)");
+    }
+
+}
+
+yarp::sig::Vector& HandPointingThread::reachablePoint(const yarp::sig::Vector& v0 , const yarp::sig::Vector& v1 , const yarp::sig::Vector& vSC, yarp::sig::Vector& vreach )
+{
+    //retrieve the coordinates of the interception between the line passing from point v0 to point v1, and a sphere with center in vSC
+
+    double exp1 = (v1[1]-v0[1])/(v1[0]-v0[0]);
+    double exp2 = (v1[2]-v0[2])/(v1[0]-v0[0]);
+    double exp3 = v0[1] - exp1*v0[0];
+    double exp4 = v0[2] - exp2*v0[0];
+
+    double d = sqrt(pow((v1[0]-v0[0]),2.0) + pow((v1[1]-v0[1]),2.0) + pow((v1[2]-v0[2]),2.0)); //object distance from robot
+    d = d>0.4 ? d : 0.4;
+    double r = d > m_reach_radius ? m_reach_radius : d;
+    
+
+    double a_eq { 1 + pow(exp1, 2.0) + pow(exp2, 2.0) };
+    double b_eq { -2*vSC[0] + 2*exp1*exp3 + 2*exp2*exp4 - 2*exp1*vSC[1] - 2*exp2*vSC[2] };
+    double c_eq { pow(vSC[0],2.0) + pow(vSC[1],2.0) + pow(vSC[2],2.0) + pow(exp3,2.0) + pow(exp4,2.0) - 2*vSC[1]*exp3 - 2*vSC[2]*exp4 - pow(r,2.0) };
+
+    vreach[0] = (-b_eq+sqrt(pow(b_eq,2.0)-4*a_eq*c_eq))/(2*a_eq);
+    vreach[1] = vreach[0]*exp1 + exp3;
+    vreach[2] = vreach[0]*exp2 + exp4;
+
+    yCInfo(HAND_POINTING_THREAD, "Reachable point defined");
+    return vreach;
+}
+
+void HandPointingThread::onRead(yarp::os::Bottle &b)
+{
+
+    yCInfo(HAND_POINTING_THREAD,"Received: %s",b.toString().c_str());
+
+    if(b.size() == 2)
+    {
+
+        //transforming the clicked point to the target end-effector position
+        double u = b.get(0).asFloat32();
+        double v = b.get(1).asFloat32();
+        yarp::sig::Vector tempPoint(4,1.0);
+
+        // We are averaging the depth value in a 10x10 box around the centre
+        double av_pixel_value = average_depth(u,v,10);
+
+        tempPoint[0] = (u - m_intrinsics.principalPointX) / m_intrinsics.focalLengthX * av_pixel_value;
+        tempPoint[1] = (v - m_intrinsics.principalPointY) / m_intrinsics.focalLengthY * av_pixel_value;
+        tempPoint[2] = av_pixel_value;
+        
+        //decide whether to use the right or left arm depending on the position of the point wrt to the torso
+        yarp::sig::Matrix transform_mtrx_torso;
+        bool torso_frame_exists = m_iTc->getTransform(m_camera_frame_id, m_torso_frame_id, transform_mtrx_torso);
```

Comment:
This transform sometimes fail. We think it might be due to race conditions on getTransform.